Core/include/Acts/Utilities/DynamicArray.hpp

```diff
@@ -0,0 +1,216 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2019 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <array>
+#include <memory>
+#include <stdexcept>
+#include <vector>
+
+namespace Acts {
+
+template <class T, std::size_t D>
+class DynamicArray {
```

Comment:
the name sounds a bit too generic to me. how about `DynamicMultidimensionalArray`

Reply:
Without looking at it in all detail, this seems similar in purpose to both `mdspan` and our already existing `Grid`. Can we use the Grid for this?

Reply:
Sounds great, salomonic decision! 🥳 We will find something new to fight over

Reply:
Dynamic Arrays replaced by the Grid. I think I can we can resolve